src/renderer/components/dock/create-resource.store.ts

```diff
@@ -1,13 +1,62 @@
+import fs from "fs-extra";
+import path from "path";
+import os from "os";
+import groupBy from "lodash/groupBy";
+import filehound from "filehound";
+import { watch } from "chokidar";
 import { autobind } from "../../utils";
 import { DockTabStore } from "./dock-tab.store";
 import { dockStore, IDockTab, TabKind } from "./dock.store";
 
 @autobind()
 export class CreateResourceStore extends DockTabStore<string> {
+
   constructor() {
     super({
       storageKey: "create_resource"
     });
+    fs.ensureDirSync(this.userTemplatesFolder);
+  }
+
+  get lensTemplatesFolder():string {
+    return path.resolve(__static, "../templates/create-resource");
+  }
+
+  get userTemplatesFolder():string {
+    return path.join(os.homedir(), ".k8slens", "templates");
+  }
+
+  get lensTemplates() {
+    return this.getTemplates(this.lensTemplatesFolder, "lens");
+  }
+
+  getTemplates(templatesPath: string, defaultGroup: string) {
+    const templates = filehound.create().path(templatesPath).ext("yaml").depth(1).findSync();
```

Comment:
This should be async.

Reply:
Also, why not also include `json` files. Also this seems like we could just use `fs.promises.readDir` and loose this extra dependency.

Reply:
> This should be async.

Why? 
It is called within the async context + I use it in getter for built-in templates, it allows to exclude re-reading built-in templates directory on any change in the user templates directory

Reply:
> Also this seems like we could just use `fs.promises.readDir` and loose this extra dependency.

https://github.com/lensapp/lens/pull/2327#discussion_r598697196 Some time ago you offered to use extra library for searching files.

Reply:
I think you are misunderstanding how async and sync play together, calling a sync function (ignoring computational functions) in an async function basically makes the async function sync.

I don't think that using a `getter` is a good enough reason to not have this being `async`. Furthermore, `getters` should be cachable and **definitely shouldn't** touch the file system or any outside dependencies. I should call that a big anti-pattern.

Reply:
> > Also this seems like we could just use `fs.promises.readDir` and loose this extra dependency.
> 
> [#2327 (comment)](https://github.com/lensapp/lens/pull/2327#discussion_r598697196) Some time ago you offered to use extra library for searching files.

I am sorry for having changing what I was suggesting. I am much more concerned about the `async` vs `sync` then about the extra dependency.

Reply:
You are right, there is some misunderstanding async programming from my side. I've fixed this one. Could you please to check it. I did some tests and it seems it works